vimrc

```diff
@@ -460,6 +460,36 @@ augroup autoformat_settings
   autocmd FileType bzl AutoFormatBuffer buildifier
 augroup END
 
+" Enable shfmt to automatically fix or format shell source code. This feature
+" can be disabled by touching .shfmt_disable in a repo's root directory.
+function s:EnableShfmt()
+  let l:cur_file_dir = expand('%:p:h')
+  if strlen(l:cur_file_dir) == 0
+    let l:git_dir = getcwd()
+  else
+    let l:git_dir = l:cur_file_dir
+  endif
+  let l:git_cmd = 'git -C ' . l:git_dir . ' rev-parse --show-toplevel 2>/dev/null'
+  let l:git_root = substitute(system(l:git_cmd), '\n\+$', '', '')
+  if v:shell_error == 0 && filereadable(l:git_root . '/.shfmt_disable')
+    augroup shells
+      autocmd!
+      autocmd FileType sh setlocal expandtab
+    augroup END
+    if has_key(g:ale_fixers, 'sh')
+      unlet g:ale_fixers.sh
+    endif
+  else
+    augroup shells
+      autocmd!
+      autocmd FileType sh setlocal noexpandtab
+    augroup END
+    let g:ale_fixers.sh = ['shfmt']
+  endif
+endfunction
```

Comment:
I don't quite understand why this function is so much more complicated than our other ways we check a file and conditionally enable ale, e.g. https://github.com/braintreeps/vim_dotfiles/blob/be345e4f21bd9e1176c02ab28ea26f79f4e6987b/vimrc#L160-L170

Reply:
Primarily because the examples you provided conditionally enable, whereas this function enables by default and conditionally disables, if the file `.shfmt_disable` is present in a repo. This function also tries to ensure that the enabling or disabling works when your current working directory is outside of the repo in which you are editing a file, e.g.

```
$ git clone foo
$ touch foo/.shfmt_disable
$ vi foo/bar.sh # shfmt will be disabled, even though you didn't start vim in the repo
```

Reply:
Could you address the first with something like
```vim
let g:ale_fixers['sh'] = ['shfmt']
 if filereadable(expand(".shfmt_disable")) 
   let g:ale_fixers['sh'] = [] 
 endif 
```

As for opening outside the repo, is that common practice? I would personally probably favor simplicity here over supporting it since we don't support it for other ale linters/fixers, and we don't have a ton of vimscript experts. But I'm not sure about other's workflows.

Reply:
With some regularity I start vim with a file that is in the sub-directory of a repository or outside the repository, so I would prefer to disable correctly in those instances.